### PR TITLE
Add auto complete

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Add auto complete. Each shell might have different setup, for `zsh`, just do:
```
ipc-agent --cli-autocomplete-gen zsh > /usr/local/share/zsh/site-functions/_ipc-agent
compinit 

# if this does not work for you, try `compdef _ipc-agent ipc-agent`
```

### CODE REVIEW

**Review by adlrocha — APPROVED** [review]
This works for me. Thanks. 

I was going to address the doc changes here, but we should change something else first for it to work. I tracked it in this issue: https://github.com/consensus-shipyard/ipc-agent/issues/203

File: src/cli/commands/mod.rs [hunk 1]
```diff
@@ -55,10 +58,13 @@ enum Commands {
 )]
 #[command(propagate_version = true)]
 struct IPCAgentCliCommands {
+    // If provided, outputs the completion file for given shell
+    #[arg(long = "generate", value_enum)]
```
Comment (adlrocha) [inline src/cli/commands/mod.rs:62]: Should we call it `install_autocompletion` or `autocompletion`? Generate seems like too general

**Review by adlrocha — CHANGES_REQUESTED** [review]
A few things to consider: 
- Change the name of the autocompletion generator flag.
- Can we add a `Makefile` rule to install autocompletion, something like `make install-autocomplete`? 
- Would you mind updating the docs to describe how to install autocompletion? Maybe we can even have a `make install` rule that does that automatically for you. 

Finally, why are you having trouble with `bash`? Let me know if I can help in any way.

**Comment by jsoares** [comment]
This warms my heart.

**Comment by jsoares** [comment]
Are we still pursuing this or should we just close?

**Comment by cryptoAtwill** [comment]
@jsoares Oh yeah, this one is pretty close to done. I will run some extra test now then should be good for review.

File: src/cli/commands/mod.rs (hunk 1 above)
Comment (cryptoAtwill) [reply src/cli/commands/mod.rs:64]: I guess this is the convention from clap?

**Review by adlrocha — APPROVED** [review]
@cryptoAtwill, if you tested it after rebasing with `dev` then LGTM. Thank you for taking thsi one to the finish line :) (it got stuck for a while haha)

**Pull request merged**